nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java

```diff
@@ -0,0 +1,160 @@
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.nucleus.microprofile.config.source;
+
+import java.beans.PropertyVetoException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.glassfish.jdbc.config.JdbcResource;
+import org.jvnet.hk2.config.ConfigBean;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.SingleConfigCode;
+import org.jvnet.hk2.config.TransactionFailure;
+import org.jvnet.hk2.config.types.Property;
+
+public class JDBCConfigSource extends PayaraConfigSource implements ConfigSource {
+
+    private static final String JDBC_CONFIG_PROPERTY_PREFIX = PROPERTY_PREFIX + "jdbc.";
+    private final JDBCConfigSourceHelper jdbcConfigHelper;
+    private final String jndiName;
+
+    public JDBCConfigSource(String jndiName) {
+        this.jndiName = jndiName;
+        this.jdbcConfigHelper = new JDBCConfigSourceHelper(getJDBCConfigProperties());
+    }
+
+    @Override
+    public Map<String, String> getProperties() {
+        return jdbcConfigHelper.getAllConfigValues();
+    }
+
+    @Override
+    public int getOrdinal() {
+        return 400;
+    }
+
+    @Override
+    public String getValue(String propertyName) {
+        return jdbcConfigHelper.getConfigValue(propertyName);
+    }
+
+    @Override
+    public String getName() {
+        return "JDBC";
+    }
+
+    private Map<String, String> getJDBCConfigProperties() {
+        HashMap<String, String> result = new HashMap<>();
+
+        if (jndiName == null) {
+            Iterator<JdbcResource> jdbcResources = domainConfiguration.getResources().getResources(JdbcResource.class).iterator();
+            while (jdbcResources.hasNext()) {
+                JdbcResource resource = jdbcResources.next();
+                putJDBCConfigProperties(result, resource.getProperty());
+            }
+            return result;
+        }
+
+        JdbcResource resource = (JdbcResource) domainConfiguration.getResources().getResourceByName(JdbcResource.class, jndiName);
+        if (resource != null) {
+            putJDBCConfigProperties(result, resource.getProperty());
+        }
+        return result;
+    }
+
+    private void putJDBCConfigProperties(Map<String, String> result, List<Property> properties) {
```

Comment:
Does this MP config source uses MP config properties to configure itself? I'm a little confused about how this Config Source is configured.